_posts/2021-02-19-26-rln-relay.md

```diff
@@ -0,0 +1,168 @@
+In traditional centralized messaging systems e.g., mail servers, spam usually signifies unsolicited messages sent in bulk or messages with malicious content like malware. Protection mechanisms include (but are not limited to) authentication through some piece of personal information e.g., phone number, checksum-based filtering (to protect against messages sent in bulk), challenge-response systems, and content filtering (on the server or via a proxy application). These methods exploit the fact that the messaging system is centralized and a global view of the users' activities is available based on which spamming patterns can be extracted and defeated accordingly.  Moreover, users are associated with an identifier (username) which enables the server to profile each user (to detect suspicious behavior like spamming) which is against the user's anonymity and privacy. 
+
+### P2P Systems
+But what about spam prevention in p2p messaging platforms?  
+there is the *Proof of Work* technique deployed by Whisper and *Peer scoring* method (namely reputation-based approach) adopted by LibP2P, but each of these solutions has its own shortcomings for real-life use-cases. 
+
+#### Proof of work
+The idea behind the proof of work is to make messaging a computationally costly operation hence lowering the messaging rate of **all** the peers including the spammers. This solution does not suit heterogeneous systems with resource-restricted peers who won't be able to carry the designated computation and will be practically excluded from the system. 
+
+#### Peer Scoring
+The peer scoring method is to limit the number of messages issued by a peer in connection to another peer. That is each peer monitors all the peers to which it is directly connected and adjusts their messaging quota (i.e., to route or not route their messages) depending on their past activities. e.g., if a peer detects its neighbor is sending more than x messages per month, can drop its quota to z.x where z is less than one (this is only one sample strategy).  One shortcoming of this solution is that scoring is based on peers' local observations and the concept of the score is defined in relation to one single peer. This leaves room for an attack where an attacker can make connections to k peers in the system and publishes k.(x-1)  messages by exploiting all of its k connections. Another attack scenario is through botnets consisting of a large number of (e.g., a million) bots. The attacker rents a botnet and inserts each of them as a legitimate peer to the network and each can publish x-1 messages per month without being identified as a spammer. <!-- Another downside is the possibility of false-positive, where the attacker can surround one single peer with some spam peers and make him look like a spammer and cause him to get removed by its neighbors. It also exhibits issues around privacy as well as arbitrary exclusion. -->
+
+
+#### Economic-Incentive Spam protection
```

Comment:
It seems odd that we put this under "alternative solutions". Maybe phrase whole thing as "Possible solutions" if we can this to be included?